sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubSchemaIOProvider.java

```diff
@@ -206,17 +211,18 @@ public Schema schema() {
       return new PTransform<PBegin, PCollection<Row>>() {
         @Override
         public PCollection<Row> expand(PBegin begin) {
+          PubsubMessageToRow.Builder builder =
+              PubsubMessageToRow.builder()
+                  .messageSchema(dataSchema)
+                  .useDlq(config.useDeadLetterQueue())
+                  .useFlatSchema(useFlatSchema);
+          if (!useFlatSchema && !fieldPresent(schema(), PAYLOAD_FIELD, FieldType.BYTES)) {
```

Comment:
Internal tests had a breakage, I think this is the reason:
```suggestion
          if (useFlatSchema || !fieldPresent(schema(), PAYLOAD_FIELD, FieldType.BYTES)) {
```

SQL PostCommit (PubSubTableProviderIT) should exercise this.

Reply:
Confirmed that this change fixes internal tests.

Reply:
fixed, thanks for catching!